sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -207,9 +206,15 @@ public Read withBigtableOptions(BigtableOptions options) {
     public Read withBigtableOptions(BigtableOptions.Builder optionsBuilder) {
       checkNotNull(optionsBuilder, "optionsBuilder");
       // TODO: is there a better way to clone a Builder? Want it to be immune from user changes.
-      BigtableOptions.Builder clonedBuilder = optionsBuilder.build().toBuilder();
-      clonedBuilder.setDataChannelCount(1);
-      clonedBuilder = addRetryOptions(clonedBuilder);
+      BigtableOptions options = optionsBuilder.build();
+      RetryOptions retryOptions = options.getRetryOptions();
+      BigtableOptions.Builder clonedBuilder = options.toBuilder()
```

Comment:
add comments with the three changes we've made:

* Need to set data channel count to 1 since there will only be 1 scanner in this session
* Need to use retryOptionsToBuilder because not in Bigtable library
* Need to override batch size because of bug.